qiskit/assembler/assemble_circuits.py

```diff
@@ -235,6 +236,44 @@ def remove_common_gate_calibrations(exps: List[QasmQobjExperiment]) -> None:
     return experiments, QasmExperimentCalibrations(gates=common_calibrations)
 
 
+def _configure_exp_los(
+    experiments: List[QasmQobjExperiment],
+    lo_converter: converters.LoConfigConverter,
+    run_config: RunConfig,
+):
+    # get per expt los
+    freq_configs = [
+        lo_converter(lo_dict) for lo_dict in getattr(run_config, "schedule_los", [])
+    ]
+
+    if len(experiments) > 1 and len(freq_configs) not in [0, 1, len(experiments)]:
+        raise QiskitError(
+            "Invalid 'schedule_los' setting specified. If specified, it should be "
+            "either have a single entry to apply the same LOs for each experiment or "
+            "have length equal to the number of experiments."
+        )
+
+    if len(freq_configs) > 1:
+        if len(experiments) > 1:
+            for idx, expt in enumerate(experiments):
+                freq_config = freq_configs[idx]
+                expt.config.qubit_lo_freq = freq_config.qubit_lo_freq
+                expt.config.meas_lo_freq = freq_config.meas_lo_freq
+        elif len(experiments) == 1:
+            expt = experiments[0]
+            experiments = []
+            for freq_config in freq_configs:
+                expt_config = copy.deepcopy(expt.config)
```

Comment:
Why is the copy required?

Reply:
If I don't include it, `expt_config` is passed by reference and so each expt gets the config from the last iteration of the loop--I tested it and removing the copy results this behavior.

The expt config only has n_qubits/memory slots at this point, so the copy should not be expensive.

Reply:
I'm not really sure why. But if I remove the `copy` and just use `expt_config = expt.config` (or even just overwrite `expt.config`) and apply that as argument to `QasmQobjExperiment` below, the last value of `expt_config` is reflected across all experiments. My guess is that the input to `QasmQobjExperiment` is done by reference, so if you pass the same config (but slightly modified), an old experiment will get the updated config. This might have to do w/ the fact that the `expt.header` and `expt.instructions` are reused. I'm not completely sure